core/src/main/java/io/grpc/Context.java

```diff
@@ -799,4 +804,29 @@ public void run() {
       listener.cancelled(Context.this);
     }
   }
+
+  /**
+   * Utility implementation of Runnable that performs the same function as {@link #wrap(Runnable)}
+   * without requiring the construction of an additional object.
+   */
+  public abstract static class ContextRunnable implements Runnable {
```

Comment:
Even though this class would be easy to maintain as part of our API, would you mind if this was just internal-only and not exposed as part of our API? Do you think there would be very many users of it?